components/camera/replaypcd/replaypcd_test.go

```diff
@@ -2,43 +2,397 @@ package replaypcd
+		replayCamCfg := &Config{Source: "source"}
+		replayCamera, _, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, false)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, "cloud connection error")
+		test.That(t, replayCamera, test.ShouldBeNil)
+	})
+
+	t.Run("bad start timestamp", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source: "source",
+			Interval: TimeInterval{
+				Start: "bad timestamp",
+			},
+		}
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeError, errors.New("invalid time format, use RFC3339"))
+		test.That(t, replayCamera, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("bad end timestamp", func(t *testing.T) {
+		replayCamCfg := &Config{
+			Source: "source",
+			Interval: TimeInterval{
+				End: "bad timestamp",
+			},
+		}
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeError, errors.New("invalid time format, use RFC3339"))
+		test.That(t, replayCamera, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+}
+
+func TestNextPointCloud(t *testing.T) {
+	t.Run("Calling NextPointCloud no filter", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{Source: "test"}
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Iterate through all files
+		for i := 0; i < numPCDFiles; i++ {
+			pc, err := replayCamera.NextPointCloud(ctx)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, pc, test.ShouldResemble, getPointCloudFromArtifact(t, i))
+		}
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
+	})
+
+	t.Run("Calling NextPointCloud with filter no data", func(t *testing.T) {
+		ctx := context.Background()
+
+		replayCamCfg := &Config{
+			Source: "test",
+			Interval: TimeInterval{
+				Start: "2000-01-01T12:00:30Z",
+				End:   "2000-01-01T12:00:40Z",
+			},
+		}
+
+		replayCamera, serverClose, err := createNewReplayPCDCamera(ctx, t, replayCamCfg, true)
+		test.That(t, err, test.ShouldBeNil)
+
+		// Confirm the end of the dataset was reached when expected
+		pc, err := replayCamera.NextPointCloud(ctx)
+		test.That(t, err, test.ShouldNotBeNil)
+		test.That(t, err.Error(), test.ShouldContainSubstring, errEndOfDataset.Error())
+		test.That(t, pc, test.ShouldBeNil)
+
+		err = replayCamera.Close(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, serverClose(), test.ShouldBeNil)
```

Comment:
These tests are great! I think we should also reduce code duplication here with table-driven tests, however. Lmk if you want to talk about what that would look like.